Modules/mathmodule.c

```diff
@@ -3479,10 +3464,6 @@ static const uint8_t factorial_trailing_zeros[] = {
 static PyObject *
 perm_comb_small(unsigned long long n, unsigned long long k, int iscomb)
 {
-    if (k == 0) {
```

Comment:
I don't think we should remove this just because it's unreachable from test cases. It's not obvious to me that `perm_comb_small` is correct without it, given potential for infinite recursion of infinite unsigned looping. If you feel strongly about removing it, probably should at least be an assert.

Reply:
> unreachable from test cases

Not just from tests...

> probably should at least be an assert

Ok.

Reply:
I guarantee that it is never 0 in the current code. In the outermost call k > 1, and in recursive calls k can be 0 only if k is 0 or 1 in the upper call, but the case k == 1 is handled before recursion by setting `fast_comb_limits2[1]` and `fast_perm_limits[1]` to `ULLONG_MAX`.

So the runtime check can be removed, but the assert can help if the code be changed in future.